inputFiles/constitutiveDrivers/scripts/pvtDriverPreprocess.py

```diff
@@ -0,0 +1,176 @@
+        componentNames = re.sub("\{|\}", "", componentNames)
+        componentNames = [s.strip() for s in componentNames.split(",")]
+
+    outputCompressibility = False
+    outputMassDensity = False
+    outputPhaseComposition = False
+    if "outputCompressibility" in driver.attrib:
+        outputCompressibility = int(driver.attrib["outputCompressibility"]) != 0
+    if "outputMassDensity" in driver.attrib:
+        outputMassDensity = int(driver.attrib["outputMassDensity"]) != 0
+    if "outputPhaseComposition" in driver.attrib:
+        outputPhaseComposition = int(driver.attrib["outputPhaseComposition"]) != 0
+
+    columns = []
+    columns.append(("Time", "seconds"))
+    columns.append(("Pressure", "Pa"))
+    columns.append(("Temperature", "K"))
+    columns.append(("Density", "kg/m3"))
+    if outputCompressibility:
+        columns.append(("Total Compressibility", "Pa^{-1}"))
+    for phaseName in phaseNames:
+        columns.append((f"Phase Fraction {phaseName}", "-"))
+    for phaseName in phaseNames:
+        columns.append((f"Phase Density {phaseName}", "kg/m3"))
+    if outputMassDensity:
+        for phaseName in phaseNames:
+            columns.append((f"Phase Mass Density {phaseName}", "kg/m3"))
+    for phaseName in phaseNames:
+        columns.append((f"Phase Viscosity {phaseName}", "Pa s"))
+    if outputPhaseComposition:
+        for phaseName in phaseNames:
+            for componentName in componentNames:
+                columns.append((f"Phase Component Fraction {phaseName} {componentName}", "-"))
+
+    return columns
+
+def loadOutputData(fileName):
+    """
+    Load the physical data from the output text file
+    
+    Args:
+        fileName (str): Path to the output text file
+
+    Returns:
+        numpy.ndarray: The data as a numpy array.
+    """
+    return numpy.loadtxt(fileName)
+
+def get_all_curves(xmlFilePath: str):
+    """
+    List all the columns reported in all output file(s)
+    
+    Args:
+        xmlFilePath (str): Path to the xml file
+
+    Returns:
+        List[str]: List of all the output columns from all PVTDriver tasks.
+    """
+    outputFiles = getOutputFilesFromXML(xmlFilePath)
+    allColumns = []
+    for driverName, _ in outputFiles:
+        columns = getColumnNames(xmlFilePath, driverName)
+        columnCount = len(columns)
+        for c in range(1, columnCount):
+            k = "%s %s" % (driverName, columns[c][0])
+            allColumns.append((k,))
+
+    return allColumns
+
+def create_hdf5_file(xmlfile: str, filename: str):
+    """
+    Creates and hdf5 file from an xml file
+    Collects all the data from the output text files and creates an hdf5 file. The data is split into
+    the different properties.
+
+    Args:
+        xmlfile (str): Path to the input xml file
+        fileName (str): Path to the output hdf5 file. This will be overwritten
+
+    Returns:
+        geos.hdf5_wrapper: The hdf5 wrapper for the data
+    """
+    
```

Comment:
are you doing this so that you can then use a `curve_check` ? This is good for now. In the future we may want to simply output an hdf5 directly from the constitutiveDriver.